cg/meta/transfer/external_data.py

```diff
@@ -144,43 +144,51 @@ def add_files_to_bundles(
             LOG.info("Adding path %s to bundle %s in housekeeper" % (path, lims_sample_id))
             self.housekeeper_api.add_file(path=path, version_obj=last_version, tags=HK_FASTQ_TAGS)
 
+    def get_failed_samples(self, fastq_paths_to_add: List[Path]) -> List[Path]:
+        failed_sums: List[Path] = []
+        for file in fastq_paths_to_add:
+            pass_check = self.check_fastq_md5sum(file)
+            if pass_check:
+                failed_sums.append(pass_check)
+        return failed_sums
+
+    def get_fastq_paths_to_add(
+        self, customer: str, hk_version: Version, lims_sample_id: str
+    ) -> List[Path]:
+        paths: List[Path] = self.get_all_paths(lims_sample_id=lims_sample_id, customer=customer)
+        fastq_paths_to_add: List[Path] = self.housekeeper_api.check_bundle_files(
+            file_paths=paths,
+            bundle_name=lims_sample_id,
+            last_version=hk_version,
+            tags=HK_FASTQ_TAGS,
+        )
+        return fastq_paths_to_add
+
     def add_transfer_to_housekeeper(self, ticket_id: int, dry_run: bool = False) -> None:
         """Creates sample bundles in housekeeper and adds the available files corresponding to the ticket to the
         bundle"""
-        failed_files: List[Path] = []
+        failed_paths: List[Path] = []
         cust: str = self.status_db.get_customer_id_from_ticket(ticket_id=ticket_id)
         available_samples: List[models.Sample] = self.get_available_samples(
             folder=self.get_destination_path(customer=cust), ticket_id=ticket_id
         )
-        cases_to_start = []
+        cases_to_start: list[dict] = []
         for sample in available_samples:
             cases_to_start.extend(
                 self.status_db.cases(sample_id=sample.internal_id, exclude_analysed=True)
             )
-
-        for sample in available_samples:
             lims_sample_id: str = sample.internal_id
```

Comment:
```suggestion
```
I think the renaming is unnecessary. I think you can just use: "sample.internal_id" everywhere instead